rules/utils/need-add-parentheses-to-member-expression-object.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const {isNotOpeningParenToken, isNotClosingParenToken} = require('eslint-utils');
+
+/**
+Check if parentheses needs to be added to a `node` when it's used as an `object` of `MemberExpression`.
+
+@param {Node} node - The AST node to check.
+@param {SourceCode} sourceCode - The source code object to get text.
+@returns {boolean}
+*/
+function needAddParenthesesToMemberExpressionObject(node, sourceCode) {
+	switch (node.type) {
+		// This is not a full list. Some other nodes like `FunctionDeclaration` don't need parentheses,
+		// but it's not possible to be in the place we are checking at this point.
+		case 'Identifier':
+		case 'MemberExpression':
+		case 'CallExpression':
+		case 'ChainExpression':
+		case 'TemplateLiteral':
+			return false;
+
+		case 'NewExpression': {
+			// `new Foo` and `new (Foo)` need add `()`
+			if (node.arguments.length === 0) {
+				const [maybeOpeningParenthesisToken, maybeClosingParenthesisToken] = sourceCode.getLastTokens(node, 2);
+				if (isNotOpeningParenToken(maybeOpeningParenthesisToken) || isNotClosingParenToken(maybeClosingParenthesisToken)) {
+					return true;
+				}
+			}
+
+			return false;
+		}
+
+		case 'Literal': {
+			/* istanbul ignore next */
+			if (typeof node.value === 'number' && /^\d+$/.test(node.raw)) {
```

Comment:
Won't this fail on `1.1` or `1e+2` and more? If this is intentional, it would be good with a code comment.

Reply:
Yes, `1.1.toString();` an  `2e+2.toString();` are  valid.

Reply:
Found 
a  better on in eslint core, https://github.com/eslint/eslint/blob/cc4871369645c3409dc56ded7a555af8a9f63d51/lib/rules/no-extra-parens.js#L1011, I didn't consider numberic seperators.

[`no-extra-parens` rule use the same pattern in initial commit.](https://github.com/eslint/eslint/commit/89a163062cdcd1cf073530c58ed8ff411cf81d11#diff-df54692ec82456195837b4983ce1f0b16b3fef0b7f8956fc2d07fcba0e4059dfR187) 😄